donjo-app/models/Analisis_indikator_model.php

```diff
@@ -124,19 +124,27 @@ public function list_data($o=0, $offset=0, $limit=500)
 		return $data;
 	}
 
+	private function validasi_data($post)
+	{
+	$data = array();
+    $data['nomor'] = bilangan($post['nomor']);
+    $data['pertanyaan'] = htmlentities($post['pertanyaan']);
```

Comment:
Periksa indentasi. Juga harus periksa semua data di form yg perlu disimpan harus ditambahkan ke `$data`. Misalnya dalam kasus ini, penyimpanan analisis indikator error karena `id_kategori` belum diambil.

Mohon periksa setiap form, ujicoba memasukkan data baru dan update data untuk meyakinkan semua berjalan normal.

Reply:
sebenarnya di fungsi `insert `dan `update` sudah ada statement `$this->input->post()` yang mengambil semua value termasuk `id_kategori`